packages/rest/src/rest.server.ts

```diff
@@ -884,18 +884,25 @@ export class RestServer extends Context implements Server, HttpServerLike {
     this._httpServer = undefined;
   }
 
-  protected _onUnhandledError(req: Request, res: Response, err: Error) {
+  protected _onUnhandledError(
+    req: Request,
+    res: Response,
+    err: Error,
+    next: Function,
+  ) {
     if (!res.headersSent) {
-      res.statusCode = 500;
-      res.end();
+      // The error can be thrown from cors middleware
+      // See https://github.com/strongloop/loopback-next/issues/4530
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      res.status((err as any).statusCode || 500).send({error: err});
+      return;
     }
 
-    // It's the responsibility of the Sequence to handle any errors.
-    // If an unhandled error escaped, then something very wrong happened
-    // and it's best to crash the process immediately.
-    process.nextTick(() => {
-      throw err;
-    });
+    // https://expressjs.com/en/guide/error-handling.html
+    // So when you add a custom error handler, you must delegate to the
+    // default Express error handler, when the headers have already been sent
+    // to the client:
```

Comment:
Please note that our custom error handler is specifically designed to detect the situation when the headers have been already sent, therefore this comment does not apply to a typical LB4 application.

Error handling in Express is a bit messy and difficult to get right in the sense that the server should return the same HTTP response body regardless of where the error originates from, don't leak any sensitive information and yet provide helpful information in debug mode. That's why we have carefully designed our own error handling schema, both in LB3 and LB4.

Let's make sure we preserve the benefits of the current design.

Reply:
Now the unexpected error is handled by the same Express error handler for CORS.